packages/react-router/src/fileRoute.ts

```diff
@@ -154,34 +144,27 @@ export class FileRoute<
   createRoute = <
     TSearchSchemaInput extends RouteConstraints['TSearchSchema'] = {},
     TSearchSchema extends RouteConstraints['TSearchSchema'] = {},
-    TSearchSchemaUsed extends Record<
-      string,
-      any
-    > = TSearchSchemaInput extends SearchSchemaInput
+    TSearchSchemaUsed = TSearchSchemaInput extends SearchSchemaInput
```

Comment:
For type parameters which act more like variables, its better not to use generic constraints on them. This is because typescript will do each checking on the value you assign to it.

Reply:
should we remove the `RouteConstraints` type then fully?

Reply:
I don't want to say constraints are bad because you need them to check generics inferred by the user

Reply:
But if it's not inferred by the user, I assume it can be trusted